SwiftReflector/Inventory/ClassInventory.cs

```diff
@@ -22,12 +22,14 @@ public override void Add (TLDefinition tld, Stream srcStm)
 				return;
 			SwiftName className = ToClassName (tld);
 			SwiftClassName formalName = ToFormalClassName (tld);
-			ClassContents contents = null;
-			if (!values.TryGetValue (className, out contents)) {
-				contents = new ClassContents (formalName, sizeofMachinePointer);
-				values.Add (className, contents);
+			lock (valuesLock) {
```

Comment:
Same as with the ErrorHandler, we could use https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-8.0 

Looking at the code on which you are locking, you could remove the need to the TryGetValue, use TryAdd which will add only if not present. If you were to update, TryUpdate would be the method you want. 


I don't know the code base, but you might be in a situation in which some Invetories are thread-safe and others are not.

Reply:
opened an issue for it. https://github.com/xamarin/binding-tools-for-swift/issues/800